src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -962,6 +967,49 @@ public static void logException(Logger logger, Level logLevel, String message, T
         }
     }
 
+    private static final String EXCEPTIONS_BUFFER_CAPACITY_ENV_VAR = "DD_JENKINS_EXCEPTIONS_BUFFER_CAPACITY";
+    private static final int DEFAULT_EXCEPTIONS_BUFFER_CAPACITY = 100;
+    private static final BlockingQueue<Pair<Date, Throwable>> EXCEPTIONS_BUFFER;
+
+    static {
+        int bufferCapacity = getExceptionsBufferCapacity();
+        if (bufferCapacity > 0) {
+            EXCEPTIONS_BUFFER = new ArrayBlockingQueue<>(bufferCapacity);
+        } else {
+            EXCEPTIONS_BUFFER = null;
+        }
+    }
+
+    private static int getExceptionsBufferCapacity() {
+        String bufferCapacityString = System.getenv("EXCEPTIONS_BUFFER_CAPACITY_ENV_VAR");
+        if (bufferCapacityString == null) {
+            return DEFAULT_EXCEPTIONS_BUFFER_CAPACITY;
+        } else {
+            try {
+                return Integer.parseInt(bufferCapacityString);
+            } catch (NumberFormatException e) {
+                severe(logger, e, EXCEPTIONS_BUFFER_CAPACITY_ENV_VAR + " environment variable has invalid value");
+                return 0;
```

Comment:
```suggestion
                return DEFAULT_EXCEPTIONS_BUFFER_CAPACITY;
```

Maybe instead of zero?